snakemake/executors/__init__.py

```diff
@@ -109,6 +109,22 @@ def get_default_resources_args(self, default_resources=None):
         default_resources = default_resources or self.workflow.default_resources
         return format_cli_arg("--default-resources", default_resources.args)
 
+    def get_resource_scopes_args(self):
+        return format_cli_arg(
+            "--set-resource-scopes", self.workflow.overwrite_resource_scopes
+        )
+
+    def get_resource_declarations(self, job):
+        resources = [
+            f"{resource}={value}"
+            for resource, value in job.resources.items()
+            if isinstance(value, int)
+            # need to check bool seperately because bool is a subclass of int
+            and not isinstance(value, bool)
```

Comment:
Ok, so isn't this too late? And in particular just silently skipping them does not sound right. I think bools as resources should cause an error beforehand, where they are specified. I.e. we should not allow them to be passed to the `snakemake()` API, and also not allow them to be present after resource expansion in `Rule.expand_resources`.

Reply:
Was just looking more into this, I made a mistake above: `parse_resources()`, and by extension `--resources` only allows `int`, not `int` or `str`, which is why currently all the `str` resources are being filtered out in `get_resource_declarations()`. 

I think passing in booleans to `--resources` will have unintended consequences on resource reckoning in the sub-workflow. Because `True` looks just like the integer `1`, snakemake will suddenly limit jobs requesting this boolean resource to one at a time, which would not be the user's intent. For example, in the test above, if 20 instances of `rule a` were being run on cluster as part of a group job with connected components, only one instance of `rule a` would run at a time because `shouldfail == True == 1`, and in the subprocess, `--resources shouldfail=true == 1`. But this limitation is not intended.

Note that this doesn't apply to `--set-resources` and `--default-resources`, which currently allow `str`, `int`, and even python expressions (for default-resources), but neither of those flags are touched by anything in this PR. 

So I would say the status quo is probably best. One other formulation could be:
```python
if isinstance(value, int)
and str(value).isdigit()
```
(or adapted as in [this SO](https://stackoverflow.com/questions/1265665/how-can-i-check-if-a-string-represents-an-int-without-using-try-except#1265696)). This would remove the specific mention of the bool class while ensuring that only "stringafiable" ints are passed on as resource limits.

Reply:
> So I would say the status quo is probably best. One other formulation could be:
> 
> ```python
> if isinstance(value, int)
> and str(value).isdigit()
> ```
> 
> (or adapted as in [this SO](https://stackoverflow.com/questions/1265665/how-can-i-check-if-a-string-represents-an-int-without-using-try-except#1265696)). This would remove the specific mention of the bool class while ensuring that only "stringafiable" ints are passed on as resource limits.

Good points. I agree, only passing real ints as limits makes most sense. The proposed solution looks good to me.